Sources/Basics/Sandbox.swift

```diff
@@ -73,8 +74,13 @@ fileprivate func macOSSandboxProfile(
     }
     // Optionally allow writing to temporary directories (a lot of use of Foundation requires this).
     else if strictness == .writableTemporaryDirectory {
-        writableDirectoriesExpression.append("(subpath \"/private/tmp\")")
-        if let tmpDir = try? TSCBasic.determineTempDirectory() {
+        // Add the standard and Foundation temporary directories, and the one determined by TSC (which also taked into account environment variables).
+        var temporaryDirectories = Set([AbsolutePath("/tmp"), AbsolutePath(NSTemporaryDirectory())])
+        if let tscTmpDir = try? TSCBasic.determineTempDirectory() {
```

Comment:
It's not clear that we should still be doing this instead of just the standard paths.

Reply:
Since the sandbox always applies to a subprocess, the most correct thing to do here is probably for the code in `Process` that applies the sandbox to look in the environment of the process it is about to launch (however it gets that environment, whether by inheritance or direct assignment, and if `TMPDIR` or any of the other usual suspects is set, then it should add that as a writable directory when it applies the sandbox.  The environment of the parent doesn't necessarily get inherited and so should ideally not affect the sandbox.

Reply:
Can the sandbox of a subprocess be more lenient than the one of the parent?

Reply:
Nested sandboxing is actually completely unsupported right now (on Darwin).  What I meant by lenient is to allow both the NSTemporaryDirectory and also any path provided by TMPDIR.  But I think we should probably remove the TMPDIR part later and just allow /tmp and NSTemporaryDirectory.  I'll do that in a later PR (we want to merge this to address CI issues).

Reply:
Adjusted it here https://github.com/apple/swift-package-manager/pull/3920